File: 2915-count-of-interesting-subarrays/2915-count-of-interesting-subarrays.java
class Solution {
    public long countInterestingSubarrays (List <Integer> nums, int modulo, int k) {
        long total = 0;
        Map <Integer, Integer> map = new HashMap <> ();
        map.put (0, 1);
        int count = 0;
        for (int num : nums) {
            if (num % modulo == k) count++;
            int remainder = (count - k) % modulo;
            if (remainder < 0) remainder += modulo;
            total += map.getOrDefault (remainder, 0);
            map.put (count % modulo, map.getOrDefault (count % modulo, 0) + 1);
        }
        return total;
    }
}

File: 2915-count-of-interesting-subarrays/2915-count-of-interesting-subarrays.c
long long countInterestingSubarrays (int* nums, int numsSize, int modulo, int k) {
    long long ans = 0;
    int count = 0;
    int *rmdMap = (int*) calloc (modulo, sizeof (int));
    rmdMap [0] = 1;
    for (int i = 0; i < numsSize; ++i) {
        count += (nums [i] % modulo == k);
        int current_remainder = count % modulo;
        int target = (current_remainder - k + modulo) % modulo;
        ans += rmdMap [target];
        rmdMap [current_remainder]++;
    }
    free (rmdMap);
    return ans;
}